### PR TITLE
feat: cumulative category for Stars Chart

### LINKED ISSUE #3025: Feature: Repo stars prototype chart
### Type of feature

🍕 Feature

### Current behavior

N/A

### Suggested solution

We want to create a chart for the repo-pages that shows a repository's stars in both a per day and cumulative version.
image

image

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by zeucapua**
## Description

Implements a "cumulative" category for the Stars Chart, where it shows the repository's total each day (eg Day 1: 1200, Day 2: 1208, etc.)

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes #3025 

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/20603494/46b9d417-13c6-4f47-9dd9-ce51cc00b375

## Tier (staff will fill in)

- [ ] Tier 1
- [x] Tier 2
- [ ] Tier 3
- [ ] Tier 4

### CODE REVIEW

**Review by jpmcb — COMMENTED** [review]
I think I found a really bizzare bug ... which I _think_ was present when this merged originally and is now more obvious with the line graph cumulitive over time: looks like any time I click to a different tab, the chart completely flips:

line-chart-bug

Very strange. Not sure what would cause that.

**Comment by bdougie** [comment]
> I think I found a really bizzare bug

I actually thought was intentional. I do admit the flippening of the chart is weird, but it does draw my attention.

**Review by bdougie — APPROVED** [review]
The cumulative chart works as expected. It would be worth fixing that chart flipping in this or a follow PR.

**Comment by jpmcb** [comment]
It doesn't make alot of sense when it flips: for example, in my gif, it shows the repo has accumulated 500+ stars. And then days latter drops to 0. Here's a more dramatic example that appears very rough:

Normal looks great:

Screenshot 2024-04-01 at 4 13 11 PM

Flipped is confusing:

Screenshot 2024-04-01 at 4 13 17 PM

it look like kubernetes went from 210k+ stars (over 2x what they actually have) down to 0 stars suddenly.

I did confirm this is in prod on the smaller "stars per day" chart so there must be something existing in the data. We can fix the flippening in a followup pr 👍🏼 📈

**Review by jpmcb — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]
📈

File: lib/utils/repo-page-utils.ts [hunk 1]
```diff
@@ -45,3 +45,59 @@ export function getStarsHistogramToDays({ stats, range }: { stats: StatsType[] |
...
+
+    previousCount = count;
+    currentTotal -= current.star_count!;
+    return days;
+  }, {});
+
+  // convert to array
+  const result: StatsType[] = [];
+  currentTotal = total;
+  for (let i = 0; i < range; i++) {
```
Comment (NsdHSO) [inline lib/utils/repo-page-utils.ts:91]: why don't you use forEach  `allDays.forEach`?

File: pages/s/[org]/[repo]/index.tsx [hunk 2]
```diff
@@ -72,7 +72,7 @@ export default function RepoPage({ repoData, image }: { repoData: DbRepo; image:
           <MetricCard variant="stars" stats={starsData} />
           <MetricCard variant="forks" stats={forkStats} />
         </section>
-        <StarsChart stats={starsData} range={range} syncId={syncId} />
+        <StarsChart stats={starsData} total={repoData.stars} range={range} syncId={syncId} />
```
Comment (NsdHSO) [inline pages/s/[org]/[repo]/index.tsx:75]:
```suggestion
        <StarsChart stats={starsData} total={repoData?.stars} range={range} syncId={syncId} />
```

**Review by NsdHSO — APPROVED** [review]
LGTM

**Pull request merged**